File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/AirCon.java
/*
 * Copyright (c) 2018 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models;

import android.support.annotation.NonNull;

import java.io.Serializable;
import java.util.Arrays;

public class AirCon implements Serializable {
    /** 指定可能は値範囲の情報 */
    public Range range;
    /**
     * 温度の単位
     * c:摂氏
     * f:華氏
     */
    public String tempUnit;

    @NonNull
    @Override
    public String toString() {
        return "AirCon{" + "range=" + range + ", tempUnit='" + tempUnit + '\'' + '}';
    }

    public static class Range {
        /** 各モード */
        public Modes modes;
        public String[] fixedButtons;

        @NonNull
        @Override
        public String toString() {
            return "Range{" + "modes=" + modes + ", fixedButtons=" + Arrays.toString(fixedButtons) + '}';
        }
    }

    public static class Modes implements Serializable {
        /** 冷房 */
        public AirConRangeMode cool;
        /** 暖房 */
        public AirConRangeMode warm;
        /** ドライ */
        public AirConRangeMode dry;
        /** 送風? */
        public AirConRangeMode blow;
        /** オート */
        public AirConRangeMode auto;

        @NonNull
        @Override
        public String toString() {
            return "Modes{" + "cool=" + cool + ", warm=" + warm + ", dry=" + dry + ", blow=" + blow + ", auto=" + auto + '}';
        }
    }

}

File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/AirCon.kt
/*
 * Copyright (c) 2019 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fishwaffle.natureremo.controller.NatureRemo.isIgnoreUnknown
import java.io.Serializable

/**
 * @param range 指定可能は値範囲の情報
 * @param tempUnit 温度の単位
 *                  c:摂氏
 *                  f:華氏
 */
@JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
data class AirCon(var range: Range?, var tempUnit: String?) : Serializable {
    /**
     * @param modes 各モード
     * @param fixedButtons
     */
    @JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
    data class Range(var modes: Modes?, var fixedButtons: List<String>?) : Serializable

    /**
     * @param cool 冷房
     * @param warm 暖房
     * @param dry ドライ
     * @param blow 送風
     * @param auto オート
     */
    @JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
    data class Modes(var cool: AirConRangeMode?, var warm: AirConRangeMode?, var dry: AirConRangeMode?, var blow: AirConRangeMode?, var auto: AirConRangeMode?) : Serializable

}



